x-pack/platform/packages/shared/kbn-content-packs-schema/src/helpers/index_pattern.ts

```diff
@@ -110,7 +111,7 @@ function locateConfiguration(
   }
 
   if (content.type === 'dashboard') {
-    const attributes = content.attributes as DashboardAttributes;
+    const attributes = content.attributes as Writable<DashboardSavedObjectAttributes>;
```

Comment:
`Writable` drops the `readonly` modifiers on the type to allow mutating the attributes.